app/services/analytics_service/src/controllers/analytics_controller.py

```
@@ -0,0 +1,565 @@
+                cached_metrics['from_cache'] = True
+                return jsonify(cached_metrics)
+            
+            # Get current time for calculations
+            now = datetime.now(timezone.utc)
+            today_start = now.replace(hour=0, minute=0, second=0, microsecond=0)
+            last_hour = now - timedelta(hours=1)
+            last_24h = now - timedelta(hours=24)
+            
+            # Get metrics from various sources
+            total_users = cache_service.get_counter("users:total") or 0
+            active_sessions = cache_service.get_counter("sessions:active") or 0
```

Pylint (3.3.7)

[warning] Redefining name 'active_sessions' from outer scope (line 36)

(W0621)